runelite-client/src/main/java/net/runelite/client/plugins/uicustomizer/UICustomizerPlugin.java

```diff
@@ -0,0 +1,291 @@
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.SpriteManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+@Slf4j
+@PluginDescriptor(
+	name = "UI Customizer",
+	enabledByDefault = false
+)
+public class UICustomizerPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private UICustomizerConfig config;
+
+	@Inject
+	private SpriteManager spriteManager;
+
+	@Provides
+	UICustomizerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(UICustomizerConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		loadSkin();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		restoreDimensions();
+		removeSkin();
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged config)
+	{
+		if (config.getGroup().equals("uiCustomizer"))
+		{
+			loadSkin();
+		}
+	}
+
+	@Subscribe
+	public void onResizableChanged(ResizeableChanged event)
+	{
+		adjustDimensions();
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		adjustDimensions();
+	}
+
+	private void loadSkin()
+	{
+		removeSkin();
+
+		Map<Integer, SpritePixels> overrides = new HashMap<>();
+
+		File sprites = new File(getClass().getResource(config.skin().toString()).getFile());
+		File[] files = sprites.listFiles();
+
+		if (files != null)
+		{
+			for (File file : files)
+			{
+				if (!file.isFile())
+				{
+					continue;
+				}
+
+				String name = file.getName();
+				name = name.substring(0, name.length() - 4);
+
+				try
+				{
+					SpritePixels pixels = getSpritePixels(file.getName());
+
+					if (name.equals(String.valueOf(SpriteID.COMPASS_TEXTURE)))
+					{
+						client.setCompass(pixels);
+					}
+					else
+					{
+						overrides.put(Integer.parseInt(name), pixels);
+					}
+				}
+				catch (NumberFormatException exception)
+				{
+					log.debug("Invalid format: " + exception);
+				}
+			}
+
+			client.setSpriteOverrides(overrides);
+		}
+
+		setupWidgets();
+	}
+
+	private void setupWidgets()
+	{
+		Map<Integer, SpritePixels> widgetOverrides = new HashMap<>();
+
+		for (WidgetOverride widgetOverride : WidgetOverride.values())
+		{
+			if (widgetOverride.getSkin() == config.skin())
+			{
+				SpritePixels spritePixels = getSpritePixels(getPath(widgetOverride));
+
+				for (WidgetInfo widgetInfo : widgetOverride.getWidgetInfo())
+				{
+					widgetOverrides.put(widgetInfo.getPackedId(), spritePixels);
+				}
+			}
+		}
+
+		client.setWidgetSpriteOverrides(widgetOverrides);
+		adjustDimensions();
+	}
+
+	private SpritePixels getSpritePixels(String path)
+	{
+		synchronized (ImageIO.class)
+		{
+			try
+			{
+				log.debug("Loading skin: " + config.skin().toString() + " - file: " + path);
+				BufferedImage image = ImageIO.read(getClass().getResourceAsStream(config.skin().toString()
+					+ "\\" + path));
+				return fileToSpritePixels(image);
```

Comment:
I would avoid holding the lock here

Reply:
What would be the best way to avoid this?